### ISSUE TITLE
TSP example infinite loop with certain data & GLPK solver

### ISSUE THREAD

**Issue by brianpiper** [JuliaOpt/JuMP.jl#194]
A lot of context here. I'm still digging around, but I wanted to see if anyone had any insight. tl;dr, try the `tsp.jl` example with the distance matrix given below.

I modified the TSP example (`tsp.jl`) to take in distance matrices for something I'm working on, and I'm using GLPK to solve. Certain instances don't solve, but loop endlessly. Here's the data:

```
[[0,11388,7838,11146,5599,11562,11615,5449,8047],
 [9416,0,2138,342,6200,5278,1154,4413,2356],
 [7809,2566,0,2319,5689,5340,3128,2972,693],
 [10059,844,2201,0,5956,5201,1493,4376,2320],
 [5373,6549,6215,6306,0,11106,6775,6859,6433],
 [8640,4841,5009,5526,10547,0,5080,3718,5227],
 [9979,876,2702,1218,7076,5508,0,4976,2920],
 [5449,5214,2833,4966,6867,4166,5775,0,2805],
 [8262,2854,1293,2831,7154,5032,3094,3048,0]]
```

Yeah, sorry it's not symmetric. I've just been pretending that it is, which I don't think is a problem, since the objective function in `tsp.jl` only grabs the distances from the upper-triangle of the matrix.

What appears to happen is that GLPK returns solutions with values that are 0.5. For example, x[1,5] = 0.5, x[1,8] = 0.5 and x[1,3] = 1.0. The code doesn't recognize the 0.5 values as arcs used by the subtour, so it loops endlessly. I initially changed line 70 from

``` julia
                if sol[cur_city, j] >= 1 - 1e-6
```

to 

``` julia
                if sol[cur_city, j] >= 1e-6
```

but, at least for the above distances and others, this just results in the same lazy constraint being added with each callback, so that didn't work, infinite loop still.

What I did is pretty.... hacky.... (`findSubtour` function)

``` julia
    while true
        # Find next node that we haven't yet visited
        found_city = false
        if rand() < 0.5
            range = n:-1:1
        else
            range = 1:n
        end
        for j in range
```

So basically, follow the other potential subtour when x[1,5] = x[1,8] = 0.5. According to a separate, standalone CPLEX tsp solver, this gets an optimal solution.

Not sure where the issue is here. My experience with lazy constraints & callbacks is spotty, but it seems like GLPK (or the julia interface?) is applying the callback at the wrong time in the solution process, or at least differently than `tsp.jl` expected.
